pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -354,6 +366,32 @@ type KubernetesSDConfig struct {
 	Selectors []K8SSelectorConfig `json:"selectors,omitempty"`
 }
 
+// ValidateSelectorRoles validate the roles based on main role.
```

Comment:
move this to https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/resource_selector.go